File: Bundles/ios_house_detail/FHHouseDetail/Classes/Detail/View/OldCells/FHDetailSuggestTipCell.m
//
//  FHDetailSuggestTipCell.m
//  FHHouseDetail
//
//  Created by 张元科 on 2019/2/15.
//

#import "FHDetailSuggestTipCell.h"
#import <Masonry.h>
#import "UIFont+House.h"
#import <UIImageView+BDWebImage.h>
#import "FHCommonDefines.h"
#import "FHDetailOldModel.h"
#import "FHURLSettings.h"
#import "TTRoute.h"
#import "UILabel+House.h"
#import "FHDetailStarsCountView.h"
#import "FHUtils.h"

@interface FHDetailSuggestTipCell ()

//@property (nonatomic, strong)   UIView       *tipBgView;
@property (nonatomic, strong)   UILabel       *tipLabel;
@property (nonatomic, strong)   UILabel       *subtitleLabel;
@property (nonatomic, weak) UIImageView *shadowImage;
@property (nonatomic, strong)   UIImageView       *trendIcon;
@property (nonatomic, strong)   UIView       *bgView;
@property(nonatomic, strong) FHDetailStarsCountView *starView;

@end

@implementation FHDetailSuggestTipCell

- (void)awakeFromNib {
    [super awakeFromNib];
    // Initialization code
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated];

    // Configure the view for the selected state
}

- (void)refreshWithData:(id)data {
    if (self.currentData == data || ![data isKindOfClass:[FHDetailSuggestTipModel class]]) {
        return;
    }
    self.currentData = data;
    //
    
    FHDetailSuggestTipModel *model = (FHDetailSuggestTipModel *)data;
    self.trendIcon.hidden = YES;
    if (model.buySuggestion && model.buySuggestion.type.length > 0) {
        self.trendIcon.hidden = NO;
        self.subtitleLabel.text = model.buySuggestion.content;
    }
    if (model.buySuggestion.score.integerValue > 0) {
        [self.starView updateStarsCountWithoutLabel:model.buySuggestion.score.integerValue];
        self.starView.hidden = NO;
    }else {
        [self.starView updateStarsCountWithoutLabel:0];
        self.starView.hidden = YES;
    }
}


-(instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier
{
    self = [super initWithStyle:style
                reuseIdentifier:reuseIdentifier];
    if (self) {
        [self setupUI];
    }
    return self;
}

- (NSString *)elementTypeString:(FHHouseType)houseType {
    return @"trade_tips";
}

- (UIImageView *)shadowImage {
    if (!_shadowImage) {
        UIImageView *shadowImage = [[UIImageView alloc]init];
        shadowImage.image = [[UIImage imageNamed:@"left_bottom_right"]resizableImageWithCapInsets:UIEdgeInsetsMake(30,30,30,30) resizingMode:UIImageResizingModeStretch];
        [self.contentView addSubview:shadowImage];
        _shadowImage = shadowImage;
    }
    return  _shadowImage;
}

- (void)setupUI {
    [self.shadowImage mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.mas_equalTo(self.contentView).offset(-20);
        make.right.mas_equalTo(self.contentView).offset(20);
        make.top.equalTo(self.contentView);
        make.bottom.mas_equalTo(self.contentView).offset(30);
    }];
    _bgView = [[UIView alloc] init];
    _bgView.backgroundColor = [UIColor themeGray7];
    [self.contentView addSubview:_bgView];
    
    _trendIcon = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"sentiment-satisfied-material"]];
    _trendIcon.contentMode = UIViewContentModeScaleAspectFill;
    [self.bgView addSubview:_trendIcon];

    _tipLabel = [UILabel createLabel:@"购房小建议" textColor:@"" fontSize:16];
    _tipLabel.textColor = [UIColor themeGray1];
    _tipLabel.font = [UIFont themeFontMedium:16];
    [self.bgView addSubview:_tipLabel];
    
    _subtitleLabel = [UILabel createLabel:@"" textColor:@"" fontSize:14];
    _subtitleLabel.textColor = [UIColor themeGray2];
    _subtitleLabel.numberOfLines = 0;
    [self.bgView addSubview:_subtitleLabel];
    
    _starView = [[FHDetailStarsCountView alloc]init];
    [self.bgView addSubview:_starView];
    
    [self.bgView mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.top.mas_equalTo(20);
        make.right.bottom.mas_equalTo(-20);
    }];
    self.bgView.layer.cornerRadius = 4;
    self.bgView.layer.masksToBounds = YES;
    
    self.trendIcon.hidden = YES;
    [self.trendIcon mas_makeConstraints:^(MASConstraintMaker *make) {
        make.top.bottom.right.mas_equalTo(self.bgView);
    }];

    [self.tipLabel mas_makeConstraints:^(MASConstraintMaker *make) {
        make.top.mas_equalTo(15);
        make.right.mas_equalTo(-15);
        make.left.mas_equalTo(15);
        make.height.mas_equalTo(22);
    }];
    [self.subtitleLabel mas_makeConstraints:^(MASConstraintMaker *make) {
        make.top.mas_equalTo(self.tipLabel.mas_bottom).offset(11);
        make.left.mas_equalTo(15);
        make.right.mas_equalTo(-15);
        make.bottom.mas_equalTo(-15);
    }];
    [self.starView mas_makeConstraints:^(MASConstraintMaker *make) {
        make.centerY.mas_equalTo(self.tipLabel);
        make.width.mas_equalTo(110);
        make.right.mas_equalTo(-20);
    }];
}

@end


// FHDetailSuggestTipModel
@implementation FHDetailSuggestTipModel

@end

File: Article/Bubble/Modules/Bizs/HorseDetailPage/Cells/FHDetailSuggestTipCell.swift
//
//  FHDetailSuggestTipCell.swift
//  Article
//
//  Created by 张静 on 2018/11/6.
//

import UIKit

class FHDetailSuggestTipCell: BaseUITableViewCell {

    
    func setBuySuggestion(suggestion: HousePriceRankSuggestion) {
        
        trendIcon.isHidden = true
        if let type = suggestion.type {
            
            subtitleLabel.text = suggestion.content
            trendIcon.isHidden = false

            if type == 1 {
                trendIcon.image = UIImage(named: "sentiment-satisfied-material")
            }else if type == 2 {
                trendIcon.image = UIImage(named: "sentiment-neutral-material")

            }else if type == 2 {
                trendIcon.image = UIImage(named: "sentiment-dissatisfied-material")
            }
        }
        
    }
    
    
    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)

        setupUI()
  
    }
    
    private func setupUI() {
        
        contentView.addSubview(bgView)
        bgView.snp.makeConstraints { maker in
            maker.left.top.equalTo(20)
            maker.right.bottom.equalTo(-20)
        }
        bgView.layer.cornerRadius = 4
        bgView.layer.masksToBounds = true
        
        trendIcon.isHidden = true
        bgView.addSubview(trendIcon)
        trendIcon.snp.makeConstraints { maker in
            maker.top.bottom.right.equalToSuperview()
        }
        
        bgView.addSubview(tipBgView)
        bgView.addSubview(tipLabel)
        
        tipBgView.snp.makeConstraints { maker in
            maker.top.equalTo(10)
            maker.left.equalTo(-4)
            maker.right.equalTo(tipLabel.snp.right).offset(12)
            maker.bottom.equalTo(tipLabel.snp.bottom).offset(3)
        }
        tipLabel.snp.makeConstraints { maker in
            maker.top.equalTo(tipBgView).offset(3)
            maker.left.equalTo(tipBgView).offset(15)
        }
        
        bgView.addSubview(subtitleLabel)
        subtitleLabel.snp.makeConstraints { maker in
            maker.top.equalTo(tipBgView.snp.bottom).offset(10)
            maker.left.equalTo(15)
            maker.right.equalTo(-15)
            maker.bottom.equalTo(-15)
        }
    }
    
    open override class var identifier: String {
        return "FHDetailSuggestTipCell"
    }
    
    lazy var tipBgView: UIView = {
        let re = UIView()
        re.backgroundColor = hexStringToUIColor(hex: kFHClearBlueColor)
        re.layer.cornerRadius = 4
        re.layer.masksToBounds = true
        return re
    }()
    
    lazy var tipLabel: UILabel = {
        let re = UILabel()
        re.textColor = .white
        re.font = CommonUIStyle.Font.pingFangRegular(16)
        re.text = "购房小建议"
        return re
    }()
    
    lazy var subtitleLabel: UILabel = {
        let re = UILabel()
        re.textColor = hexStringToUIColor(hex: kFHMutedBlueColor)
        re.font = CommonUIStyle.Font.pingFangRegular(14)
        re.numberOfLines = 0
        return re
    }()
    
    lazy var trendIcon: UIImageView = {
        let re = UIImageView()
        re.contentMode = .scaleAspectFill
        re.image = UIImage(named: "sentiment-satisfied-material")
        return re
    }()
    
    lazy var bgView: UIView = {
        let re = UIView()
        re.backgroundColor = hexStringToUIColor(hex: "#f0f8ff")
        return re
    }()
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    
    
    
    override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
    }

    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)

        // Configure the view for the selected state
    }

}

func parsePriceRangeNode(_ priceRank: HousePriceRank?, traceExtension: TracerParams = TracerParams.momoid()) -> () -> TableSectionNode? {
    return {
        
        var buySuggestion: HousePriceRankSuggestion?
        if let thePriceRank = priceRank {
            buySuggestion = thePriceRank.buySuggestion
        }
        
        var suggestion: HousePriceRankSuggestion = HousePriceRankSuggestion()
        suggestion.type = 2
        suggestion.content = "该房源当前价格合理，房东诚心出售，建议尽快联系经纪人看房"
        
        let cellRender = oneTimeRender(curry(fillPriceRangeCell)(buySuggestion ?? suggestion))
        let params = TracerParams.momoid() <|>
            toTracerParams("price_reference", key: "element_type") <|>
            toTracerParams("old_detail", key: "page_type") <|>
        traceExtension
        
        return TableSectionNode(
            items: [cellRender],
            selectors: nil,
            tracer: [elementShowOnceRecord(params: params)],
            label: "",
            type: .node(identifier: FHDetailSuggestTipCell.identifier))
        
    }
}

func fillPriceRangeCell(
    _ data: HousePriceRankSuggestion,
    cell: BaseUITableViewCell) -> Void {
    if let theCell = cell as? FHDetailSuggestTipCell {
        
        theCell.setBuySuggestion(suggestion: data)
    }
    
}


